File: dotnet/SpiderEngine/Model/SpiderInfo.cs
namespace SpiderEngine.Model;

public class SpiderInfo
{
    public string Name { get; set; } = default!;
    public int Version { get; set; }
    public IEnumerable<string> Environment { get; set; } = default!;
    public EngineOptions Engine { get; set; } = default!;
    public IEnumerable<SpiderTaskInfo> Task { get; set; } = default!;
    public IEnumerable<string> Output { get; set; } = default!;
}

File: android/src/main/kotlin/com/itoken/team/wutspider/model/SpiderInfo.kt
package com.itoken.team.wutspider.model

import kotlinx.serialization.Serializable

@Serializable
data class SpiderInfo(
    val name: String? = null,
    val version: Int = 0,
    val environment: List<String>? = null,
    val engine: EngineOptions = EngineOptions(cookie = true, forceSSL = false),
    val task: List<SpiderTaskInfo>? = null,
    val output: List<String>? = null
)